peg-macros/translate.rs

```diff
@@ -260,23 +260,48 @@ fn compile_rule(context: &Context, rule: &Rule) -> TokenStream {
             quote!()
         };
 
+        let not_cached_name = format_ident!("{}_not_cached", name);
+        let extra_args_call = &context.extra_args_call;
         quote_spanned! { span =>
-            if let Some(entry) = __state.#cache_field.get(&__pos) {
-                #cache_trace
-                return entry.clone();
+            fn #not_cached_name<'input #(, #grammar_lifetime_params)* #(, #ty_params)*>(__input: #input_ty, __state: #parse_state_ty, __err_state: &mut ::peg::error::ErrorState, __pos: usize #extra_args_def) -> ::peg::RuleResult<#ret_ty> {
```

Comment:
This could be nested inside the other function. Or, since it's only called in one place, any reason not to simply inline its body into `let __current_result = { ... }`?

Reply:
Trying to make it more explicit, but if it's clear enough let's inline it.